src/utils/Simplify.h

```diff
@@ -0,0 +1,465 @@
+     * (a polyline).
+     * \return A measure of how important the vertex is. Higher importance means
+     * that the vertex should probably be retained in the output.
+     */
+    template<typename Polygonal>
+    coord_t importance(const Polygonal& polygon, const std::vector<bool>& to_delete, const size_t index, const bool is_closed) const
+    {
+        const size_t poly_size = polygon.size();
+        if(!is_closed && (index == 0 || index == poly_size - 1))
+        {
+            return std::numeric_limits<coord_t>::max(); //Endpoints of the polyline must always be retained.
+        }
+        //From here on out we can safely look at the vertex neighbors and assume it's a polygon. We won't go out of bounds of the polyline.
+
+        const Point vertex = getPosition(polygon[index]);
+        const size_t before_index = previousNotDeleted(index, to_delete);
+        const size_t after_index = nextNotDeleted(index, to_delete);
+
+        const coord_t area_deviation = getAreaDeviation(polygon[before_index], polygon[index], polygon[after_index]);
+        if(area_deviation > max_area_deviation) //Removing this line causes the variable line width to get flattened out too much.
+        {
+            return std::numeric_limits<coord_t>::max();
+        }
+
+        const Point before = getPosition(polygon[before_index]);
+        const Point after = getPosition(polygon[after_index]);
+        const coord_t deviation2 = LinearAlg2D::getDist2FromLine(vertex, before, after);
+        if(deviation2 <= min_resolution * min_resolution) //Deviation so small that it's always desired to remove them.
+        {
+            return deviation2;
+        }
+        if(vSize2(before - vertex) > max_resolution * max_resolution && vSize2(after - vertex) > max_resolution * max_resolution)
+        {
+            return std::numeric_limits<coord_t>::max(); //Long line segments, no need to remove this one.
+        }
+        return deviation2;
+    }
+
+    /*!
+     * Mark a vertex for removal.
+     *
+     * This function looks in the vertex and the four edges surrounding it to
+     * determine the best way to remove the given vertex. It may choose instead
+     * to delete an edge, fusing two vertices together.
+     * \tparam Polygonal A polygonal object, which is a list of vertices.
+     * \param polygon The polygon to remove a vertex from.
+     * \param to_delete The vertices that have been marked for deletion so far.
+     * This will be edited in-place.
+     * \param vertex The index of the vertex to remove.
+     * \param deviation2 The previously found deviation for this vertex.
+     * \param is_closed Whether we're working on a closed polygon or an open
+     * polyline.
+     */
+    template<typename Polygonal>
+    void remove(Polygonal& polygon, std::vector<bool>& to_delete, const size_t vertex, const coord_t deviation2, const bool is_closed) const
+    {
+        if(deviation2 <= min_resolution * min_resolution)
```

Comment:
Since `min_resolution` is constexpr, this is guaranteed to be optimised by the compiler.

I think adding new variables makes it harder to follow, and changing the existing one to be squared is easy to misunderstand, so I'd rather keep it this way.